lib/apiservers/engine/backends/container.go

```diff
@@ -381,7 +407,8 @@ func (c *Container) ContainerExecResize(eid string, height, width int) error {
 // ContainerExecStart starts a previously set up exec instance. The
 // std streams are set up.
 func (c *Container) ContainerExecStart(ctx context.Context, eid string, stdin io.ReadCloser, stdout io.Writer, stderr io.Writer) error {
-	defer trace.End(trace.Begin(eid))
+	op := trace.NewOperation(ctx, "")
+	defer trace.End(trace.Begin(fmt.Sprintf("opID=(%s) eid=(%s)", op.ID(), eid)))
```

Comment:
you can add a Stringer to op and use op here instead of op.ID()

```
func (o *Operation) String() string {
	return o.id
}
```

Reply:
Ah I see. So that it has the same behavior as `error`

Reply:
I used `return o.header()` instead of `o.id` :)

Reply:
I am also changing this to using the op.Sprintf since it will tack the header on by default. I wanted to promote the use of op.Sprintf for providing strings to the `Trace.Begin` and `Trace.End` calls since I still love those and it offers a much cleaner way of adding the Operation header to that logging.